### PR TITLE
RzUtil: fix rz_path_prefix in case of portable build

### PR DESCRIPTION
**Pull request by ret2libc**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [x] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

Before this patch the function was using RZ_PREFIX even when IS_PORTABLE
was set, making the function not working well because RZ_PREFIX might
not exist in the system where the portable build is used (the path was
passed to realpath(), which requires an existing path).

This patch fixes the problem by searching for the BINDIR without using
realpath() on the RZ_PREFIX. Moreover, it avoid retrieving the same
prefix every time by saving it in a global variable.

**Test plan**

- Build with -Dportable=true, `eco` should list all themes
- Build with -Dportable=true --bindir=., `eco` should list all themes
- Build with -Dportable=false, `eco` should list all themes

To better test, spawn a clean container so that files are not loaded from user home dir or other system places.

@wargio Would this work for Cutter as well?

### CODE REVIEW

File: librz/util/path.c [hunk 1]
```diff
@@ -12,46 +12,112 @@
...
 	if (!pid_to_path) {
-		goto prefix;
+		return NULL;
 	}
-	char *bindir = rz_path_realpath(RZ_JOIN_2_PATHS(RZ_PREFIX, RZ_BINDIR));
-	if (!bindir) {
-		goto prefix;
+
+	const char *filename = rz_file_basename(pid_to_path);
+	if (rz_str_startswith(filename, "test_")) {
```
Comment (XVilka) [inline librz/util/path.c:36]: Why is that?

File: librz/util/path.c [hunk 2]
```diff
@@ -12,46 +12,112 @@
...
+#endif
+
+/**
+ * \brief Return \p path prefixed by the Rizin install prefix
+ *
+ * The install prefix is taken from the build-time configuration RZ_PREFIX,
+ * unless Rizin was not compiled as "portable". In such a case the prefix is
+ * discovered from the path of the executable calling this function.
+ *
+ * Executables starting with "test_" are considered to be part of the testing
```
Comment (XVilka) [inline librz/util/path.c:99]: Never mind, I see. Maybe duplicate comment there?

File: librz/util/path.c (hunk 1 above)
Comment (wargio) [reply librz/util/path.c:36]: i think is related to the test suite.

**Review by wargio — APPROVED** [review]

**Review by XVilka — CHANGES_REQUESTED** [review]
Windows builds should be fixed. Please also use `dist-` branch to be sure it works on all systems supported by CI.

```
FAILED: librz/util/rz_util.lib.p/signal.c.obj 
"cl" "-Ilibrz\util\rz_util.lib.p" "-I." "-I.." "-Ilibrz" "-I..\librz" "-Ilibrz\include" "-I..\librz\include" "-Isubprojects\sdb" "-I..\subprojects\sdb" "-Isubprojects\sdb\src" "-I..\subprojects\sdb\src" "-Isubprojects\zlib-1.2.12" "-I..\subprojects\zlib-1.2.12" "/MT" "/nologo" "/showIncludes" "/utf-8" "/W2" "/O2" "/Gw" "-DRZ_PLUGIN_INCORE=1" "/Fdlibrz\util\rz_util.lib.p\signal.c.pdb" /Folibrz/util/rz_util.lib.p/signal.c.obj "/c" ../librz/util/signal.c
../librz/util/signal.c(74): error C2143: syntax error: missing ')' before '*'
../librz/util/signal.c(74): error C2143: syntax error: missing '{' before '*'
../librz/util/signal.c(74): error C2143: syntax error: missing ';' before '*'
../librz/util/signal.c(74): error C2059: syntax error: ')'
../librz/util/signal.c(74): error C2054: expected '(' to follow 'oldmask'
[93/1789] Compiling C object librz/util/rz_util.lib.p/spaces.c.obj
[94/1789] Compiling C object librz/util/rz_util.lib.p/regex_regexec.c.obj
ninja: build stopped: subcommand failed.
```
```
[116/1789] Compiling C object librz/util/rz_util.lib.p/thread_sem.c.obj
FAILED: librz/util/rz_util.lib.p/thread_sem.c.obj 
"cl" "-Ilibrz\util\rz_util.lib.p" "-I." "-I.." "-Ilibrz" "-I..\librz" "-Ilibrz\include" "-I..\librz\include" "-Isubprojects\sdb" "-I..\subprojects\sdb" "-Isubprojects\sdb\src" "-I..\subprojects\sdb\src" "-Isubprojects\zlib-1.2.12" "-I..\subprojects\zlib-1.2.12" "/MT" "/nologo" "/showIncludes" "/utf-8" "/W2" "/O2" "/Gw" "-DRZ_PLUGIN_INCORE=1" "/Fdlibrz\util\rz_util.lib.p\thread_sem.c.pdb" /Folibrz/util/rz_util.lib.p/thread_sem.c.obj "/c" ../librz/util/thread_sem.c
../librz/util/thread_sem.c(39): error C2065: 'sem_t': undeclared identifier
[117/1789] Compiling C object librz/util/rz_util.lib.p/thread_lock.c.obj
FAILED: librz/util/rz_util.lib.p/thread_lock.c.obj 
"cl" "-Ilibrz\util\rz_util.lib.p" "-I." "-I.." "-Ilibrz" "-I..\librz" "-Ilibrz\include" "-I..\librz\include" "-Isubprojects\sdb" "-I..\subprojects\sdb" "-Isubprojects\sdb\src" "-I..\subprojects\sdb\src" "-Isubprojects\zlib-1.2.12" "-I..\subprojects\zlib-1.2.12" "/MT" "/nologo" "/showIncludes" "/utf-8" "/W2" "/O2" "/Gw" "-DRZ_PLUGIN_INCORE=1" "/Fdlibrz\util\rz_util.lib.p\thread_lock.c.pdb" /Folibrz/util/rz_util.lib.p/thread_lock.c.obj "/c" ../librz/util/thread_lock.c
../librz/util/thread_lock.c(13): error C2065: 'pthread_mutexattr_t': undeclared identifier
../librz/util/thread_lock.c(13): error C2146: syntax error: missing ';' before identifier 'attr'
../librz/util/thread_lock.c(13): error C2065: 'attr': undeclared identifier
../librz/util/thread_lock.c(14): error C2065: 'attr': undeclared identifier
../librz/util/thread_lock.c(16): error C2065: 'attr': undeclared identifier
../librz/util/thread_lock.c(16): error C2065: 'PTHREAD_MUTEX_RECURSIVE': undeclared identifier
../librz/util/thread_lock.c(20): error C2065: 'attr': undeclared identifier
```

**Comment by ret2libc** [comment]
https://github.com/rizinorg/rizin/pull/2646

**Pull request closed without merging**